### PR TITLE
fix #18594: Unnecessarily allocating an error string

### LINKED ISSUE #18594: Unnecessarily allocating an error string
In: https://github.com/servo/servo/blob/29517d553e6c2fddc7e3cf0c0abef58c6d0f34aa/components/script/document_loader.rs#L133

there's a line:
```rust
        self.blocking_loads.remove(idx.expect(&format!("unknown completed load {:?}", load)));
```

This allocates the error string even if the `idx` exists.

This should be replaced by `idx.unwrap_or_else(|| panic!(...))`.

### PR DESCRIPTION
**Pull request by tigercosmos**
r? @asajeffrey 

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #18594 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18603)

### CODE REVIEW

File: components/script/document_loader.rs [hunk 1]
```diff
@@ -130,7 +130,7 @@ impl DocumentLoader {
     pub fn finish_load(&mut self, load: &LoadType) {
         debug!("Removing blocking load {:?} ({}).", load, self.blocking_loads.len());
         let idx = self.blocking_loads.iter().position(|unfinished| *unfinished == *load);
-        self.blocking_loads.remove(idx.expect(&format!("unknown completed load {:?}", load)));
+        idx.unwrap_or_else(|| panic!("unknown completed load {:?}", load));
```
Comment (emilio) [inline components/script/document_loader.rs]: We're no longer removing the load?

Also, this could be a bit cleaner IMO just using `match`.

**Review by emilio — CHANGES_REQUESTED** [review]

File: components/script/document_loader.rs (hunk 1 above)
Comment (asajeffrey) [reply components/script/document_loader.rs]: Sorry, I should have been clearer, the `self.blocking_loads.remove(...)` should stay.

Thanks for taking this on!

Comment (tigercosmos) [reply components/script/document_loader.rs]: Is it fine to change to the following?
```
self.blocking_loads.remove(idx.unwrap_or_else(|| panic!("unknown completed load {:?}", load)));
```

Comment (asajeffrey) [reply components/script/document_loader.rs]: Yes, that looks good!

**Comment by KiChjang** [comment]
@bors-servo r+

Thanks!

**Comment by asajeffrey** [comment]
Yay!

**Pull request merged**